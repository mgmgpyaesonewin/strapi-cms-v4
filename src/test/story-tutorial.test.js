const axios = require('axios');
const storyTutorials = require("./api/story-tutorial/story-tutorial.js");
describe("fetchStoryTutorialList", () => {
    describe("when storyTutorials API call is successful", () => {
        it("should return story tutorials list", async () => {
            let response = await storyTutorials.fetchStoryTutorialList();
            let stories = response.data;
            expect(stories[0]).toMatchSnapshot({
                description: expect.objectContaining({
                    en: expect.any(String),
                    my: expect.any(String),
                    zw: expect.any(String)
                }),
                id: expect.any(Number),
                image: expect.any(Object),
                position: expect.any(String),
                story_name: expect.any(String),
                title: expect.objectContaining({
                    en: expect.any(String),
                    my: expect.any(String),
                    zw: expect.any(String)
                }),
                type: expect.any(String),
                // wp_tutorial_stories: expect.arrayContaining([
                // wp_tutorial_stories: expect.arrayContaining([
                //     expect.objectContaining(
                //         {
                //             id: expect.any(Number),
                //             //createdAt : expect.any(Date),
                //             //updatedAt :expect.any(Date),
                //             //publishedAt:expect.any(Date),
                //             name: expect.any(String),
                //             title: expect.objectContaining({
                //                 en: expect.any(String),
                //                 my: expect.any(String),
                //                 zw: expect.any(String),
                //                 id: expect.any(Number)
                //             }),
                //             description: expect.objectContaining({
                //                 en: expect.any(String),
                //                 my: expect.any(String),
                //                 zw: expect.any(String),
                //                 id: expect.any(Number)
                //             }),
                //             image: expect.any(Object),

                //         }
                //     ),

                // ])

            });

            expect(stories[0].wp_tutorial_stories[0]).toMatchSnapshot({
                id: expect.any(Number),
                name: expect.any(String),
                title: expect.objectContaining({
                    en: expect.any(String),
                    my: expect.any(String),
                    zw: expect.any(String),
                    id: expect.any(Number)
                }),
                description: expect.objectContaining({
                    en: expect.any(String),
                    my: expect.any(String),
                    zw: expect.any(String),
                    id: expect.any(Number)
                }),
                image: expect.any(Object),
            });

            //expect(response.data.data).toMatchSnapshot();
        });
    });
});