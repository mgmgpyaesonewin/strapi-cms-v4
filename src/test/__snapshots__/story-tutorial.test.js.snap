// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetchStoryTutorialList when storyTutorials API call is successful should return story tutorials list 1`] = `
Object {
  "description": ObjectContaining {
    "en": Any<String>,
    "my": Any<String>,
    "zw": Any<String>,
  },
  "id": Any<Number>,
  "image": Any<Object>,
  "position": Any<String>,
  "story_name": Any<String>,
  "title": ObjectContaining {
    "en": Any<String>,
    "my": Any<String>,
    "zw": Any<String>,
  },
  "type": Any<String>,
  "wp_tutorial_stories": Array [
    Object {
      "app": "merchant",
      "createdAt": "2022-06-29T06:59:30.205Z",
      "description": Object {
        "en": "story1",
        "id": 44,
        "my": "story2",
        "zw": "story2",
      },
      "id": 2,
      "image": Object {
        "url": "https://wave-strapi-staging-s3-bucket.s3.ap-southeast-1.amazonaws.com/story_Frames_ffa9dbecb4.png",
      },
      "isPinned": true,
      "name": "Story 1",
      "position": "1",
      "publishedAt": "2022-07-18T05:09:54.792Z",
      "title": Object {
        "en": "Story 1",
        "id": 19,
        "my": "Story 1",
        "zw": "Story 1",
      },
      "updatedAt": "2022-07-18T05:09:54.797Z",
    },
    Object {
      "app": "merchant",
      "createdAt": "2022-07-02T15:54:49.917Z",
      "description": Object {
        "en": "Itaque dolorem maior",
        "id": 64,
        "my": "Sunt ex qui dolorem ",
        "zw": "Dolorum magni qui il",
      },
      "id": 3,
      "image": Object {
        "url": "https://wave-strapi-staging-s3-bucket.s3.ap-southeast-1.amazonaws.com/story_Frames_ffa9dbecb4.png",
      },
      "isPinned": true,
      "name": "Hilary Sexton",
      "position": "2",
      "publishedAt": "2022-07-02T15:54:52.407Z",
      "title": Object {
        "en": "Nostrum tempor conse",
        "id": 63,
        "my": "Ipsam deserunt aute ",
        "zw": "Aliquam nisi autem s",
      },
      "updatedAt": "2022-07-18T04:25:20.120Z",
    },
  ],
}
`;

exports[`fetchStoryTutorialList when storyTutorials API call is successful should return story tutorials list 2`] = `
Object {
  "app": "merchant",
  "createdAt": "2022-06-29T06:59:30.205Z",
  "description": ObjectContaining {
    "en": Any<String>,
    "id": Any<Number>,
    "my": Any<String>,
    "zw": Any<String>,
  },
  "id": Any<Number>,
  "image": Any<Object>,
  "isPinned": true,
  "name": Any<String>,
  "position": "1",
  "publishedAt": "2022-07-18T05:09:54.792Z",
  "title": ObjectContaining {
    "en": Any<String>,
    "id": Any<Number>,
    "my": Any<String>,
    "zw": Any<String>,
  },
  "updatedAt": "2022-07-18T05:09:54.797Z",
}
`;
